### ISSUE TITLE
No bugs found, or Infer failed midway?

### ISSUE THREAD

**Issue by hmijail** [facebook/infer#699]
Infer 0.12.0.

I'm running Infer on a C project, and seemingly no problems are being found. Nothing is reported in infer's output, no bugs.txt or report.csv files are generated, and inferTraceBugs errors out.

```
$ infer analyze
Found 45 source files to analyze in /blah/infer-out
Starting analysis...

legend:
  "F" analyzing a file
  "." analyzing a procedure

FFFF................................................................................................................................................................................................................F..............................F..............F.............F...........................................F.....F........................................................................................................F............................................................................................F...F..........F...............................F...................FF........................................F.................F........................F..........F..............................F..........................................................F....F.........................FF.............FF..........F....................F.......................FF..F...FF..........F...F....F....................F........F...........F.............F...................F...............F............F...................................................................

$ inferTraceBugs
Traceback (most recent call last):
  File "/usr/local/bin/inferTraceBugs", line 399, in <module>
    main()
  File "/usr/local/bin/inferTraceBugs", line 372, in main
    reports = utils.load_json_from_path(report_filename)
  File "/usr/local/Cellar/infer/0.12.0_1/lib/infer/infer/lib/python/inferlib/utils.py", line 83, in load_json_from_path
    encoding=config.CODESET, errors=errors) as file_in:
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/codecs.py", line 896, in open
    file = __builtin__.open(filename, mode, buffering)
IOError: [Errno 2] No such file or directory: u'/blah/infer-out/report.json'
```

The lack of *any* reporting makes me wonder whether infer actually finished its job correctly. Or is this expected behavior?

If it is, I'd suggest that some explicit notification of "No problems found" would be good.

I have tried running `infer -g analyze`, but doesn't show anything I'd call interesting.

For context: my build system is not supported, so for the capture phase I've forced the builder to prefix every clang invokation with  `infer capture --reactive --`.

**Comment by jvillard** [comment]
Thanks for the report. It looks like infer silently crashed during the run. Is there anything useful in infer-out/logs?

**Comment by hmijail** [comment]
In infer-out/log/analyze, there is an err.log and an out.log, each weighting 1 and 24 MB.
There are also infer-out/log/driver and infer-out/log/capture (89 items, 16 MB). 

I could upload them all somewhere non-public, or I could anonymize somehow the logs (any suggestions?)

**Comment by hmijail** [comment]
For completeness, the output of infer -g analyze:
```
Found 45 source files to analyze in /blah/infer-out
Starting analysis...

legend:
  "F" analyzing a file
  "." analyzing a procedure
  "C" analyzer crashed
  "T" timeout: procedure analysis took too much time
  "S" timeout: procedure analysis took too many symbolic execution steps
  "R" timeout: procedure analysis took too many recursive iterations

FFFF.........................SS..S..................SS...............S.......S.............SS......S.S..S..S.SSS..............SSS...........S..S.S...S..........S........SS...S..S...S....SS...S.......SS.S...SSS.S.S..SS..S......S.S.S....SS...........F.SS..SSS.......SS.....S.S...S...S......S...........S.......S.S................S..S.......F....SSSF..S..........SF...............S....SS..S....FS.....S....F...S..S...S....S..S....SS...S.........S................S..S.......SS.........S................S.S.......S..S.S...S.....S.S.......FS....S..............SF...S.......S.S.S..........................S...............S........S.S.....S.................S...F....SS..F....................SS.S.......F......S..SS.FF....S.S.........SSF.S........S.......S.SS.....S.S.SS.S...........F........S.......F......S........SS......S.S.S...S.S..S....S.......F..S......S..........S....SS.........S....S..F.........FS........F...............................S.......S.........SF...S.....S.....SF.S.FF.S..F......S..S.........S.........S.F.....S......F.S.......FF..FF....F..........SF............S...........F............S...S.F....F...........F.S..S..F...S.S..........S........S........F.......F.........SS.SSF.S..S........S....S.S.....S.................S..S.S..S...S.S..S.............SS..S.S.S....
```

**Comment by jvillard** [comment]
If you can, it would be worth trying the latest version of infer as we've removed some (but maybe not all) of these cases where infer silently crashes. The new version also has a single log file, which should be easier to share/send by email.

**Comment by hmijail** [comment]
I'm not sure I have enough time for that, but I'll try, thank you.

**Comment by hmijail** [comment]
With a build-from-source Infer (version v0.12.0-2dcde3a8), the analysis does end generating a list of problems. So... I guess this bug can be closed?

**Comment by jvillard** [comment]
Great, thanks for the update. There have been a number of fixes since 0.12.0 that could have been related so I was hoping for that outcome.
